flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -100,7 +122,16 @@ public RemoteInputChannel(
 	}
 
 	void assignExclusiveSegments(List<MemorySegment> segments) {
```

Comment:
Yes, this method should be called only once after `RemoteInputChannel` created. I think  `assignExclusiveSegments` can describe the semantics of invoking only once, and `addExclusiveSegments` seems allow to be called multiple times.

I currently take the third way you suggested, adding the `checkState` to avoid multi calling.

Reply:
works for me :)